File: subprojects/docs/src/samples/userguide/plugins/namedDomainObjectContainer/java/buildSrc/src/main/java/Resource.java

import java.net.URI;

public class Resource {
    private final String name;
    private URI uri;
    private String userName;

    public Resource(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public URI getUri() {
        return uri;
    }

    public void setUri(URI uri) {
        this.uri = uri;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }
}

File: subprojects/docs/src/samples/userguide/plugins/namedDomainObjectContainer/kotlin/buildSrc/src/main/kotlin/Resource.kt
import java.net.URI
// tag::resource[]

// Type must have a public constructor that takes the element name as a parameter
// Type must have a 'name' property, which should be read-only
open class Resource(val name: String) {
    var uri: URI? = null
    var userName: String? = null
}
// end::resource[]
